internal/datacoord/segment_manager.go

```diff
@@ -493,7 +493,10 @@ func (s *SegmentManager) DropSegmentsOfChannel(ctx context.Context, channel stri
 	validSegments := make([]int64, 0, len(s.segments))
 	for _, sid := range s.segments {
 		segment := s.meta.GetSegment(sid)
-		if segment != nil && segment.GetInsertChannel() != channel {
+		if segment == nil {
+			continue
+		}
+		if segment.GetInsertChannel() != channel {
```

Comment:
If channel not matched, the loop should be skipped. Otherwise valid segments allocation will be truncated.

Reply:
Fixed